src/components/common/FileUploadPdf.vue

```diff
@@ -246,16 +261,31 @@ export default class FileUploadPdf extends Vue {
    * @returns the file key on success, or null on failure
    */
   async uploadFile (file: File): Promise<string> {
-    try {
-      // NB: will throw on API error
-      const psu = await this.getPresignedUrl(file.name)
+    let psu: PresignedUrlIF
+    let res
 
-      // NB: will throw on API error
-      const res = await this.uploadToUrl(psu.preSignedUrl, file, psu.key, this.userId)
+    try {
+      if (this.enableDocumentRecords) {
+        if (!this.documentClass || !this.documentType) {
+          this.errorMessages = ['An error occurred while uploading. Please try again.']
+          return null
+        }
+        res = await DocumentServices.uploadDocumentToDRS(
+          file,
+          this.documentClass,
+          this.documentType,
+          this.businessIdentifier
+        )
+      } else {
+        // NB: will throw on API error
+        psu = await this.getPresignedUrl(file.name)
 
+        // NB: will throw on API error
+        res = await this.uploadToUrl(psu.preSignedUrl, file, psu.key, this.userId)
+      }
       // check if successful
-      if (res?.status === 200) {
-        return psu.key
+      if (res && [StatusCodes.OK, StatusCodes.CREATED].includes(res.status)) {
+        return this.enableDocumentRecords ? res.data.documentServiceId : psu.key
```

Comment:
OK, cool, I like this integration of DRS and Minio.

I'm wondering how we're going to migrate / cut-over from one to the other, because this implementation doesn't support both at the same time. Thoughts?

Reply:
- The document will be uploaded to either Minio or DRS, depending on the feature flag.
- For downloading documents, a regex is used to determine the source:
     ```DRS_ID_PATTERN = /^DS\d{10}$/```
  - Example Minio file key: `adb1e8a0-5dd5-4f71-8726-6845ee3185ff.pdf`
  - Example DRS service ID: `DS0000101295`

Reply:
Thanks for explaining upload vs download. Now I get it :)